pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -650,6 +650,12 @@ type CommonPrometheusFields struct {
 	// +optional
 	// +kubebuilder:validation:Minimum=60
 	MaximumStartupDurationSeconds *int32 `json:"maximumStartupDurationSeconds,omitempty"`
+
+	// EXPERIMENTAL List of scrape classes to expose to monitors and other scrape configs.
+	// This is experimental feature and might change in the future.
+	// +listType=map
+	// +listMapKey=name
+	ScrapeClasses []ScrapeClass `json:"scrapeClasses,omitempty"`
```

Comment:
```suggestion
	ScrapeClasses []*ScrapeClass `json:"scrapeClasses,omitempty"`
```

ScrapeClass is an object that we expect to grow, a lot! If we work with value receivers, instead of pointers, I suspect that reconciling Prometheus instances with a large amount of ScrapeClasses could increase memory allocations

I explain in the comments below, but a good thread on Twitter also explains the difference of working with [Pointers vs Value receivers](https://twitter.com/MattJamesBoyle/status/1742822532372136433).

Reply:
I totally agree, btw thanks for sharing the thread 🥳